src/dmd/dmodule.d

```diff
@@ -1000,20 +987,57 @@ extern (C++) final class Module : Package
                     pkg.isPkgMod = PKG.module_;
                     pkg.mod = this;
                     pkg.tag = this.tag; // reuse the same package tag
+                    return pkg;
                     amodules.push(this); // Add to global array of all modules
                 }
                 else
+                {
+                    const(char)* srcname = srcfile.name.toChars();
                     error(md ? md.loc : loc, "from file %s conflicts with package name %s", srcname, pkg.toChars());
+                    return this;
+                }
             }
-            else
-                assert(global.errors);
+        }
+
+        Package ret = this;
+        if (isPackageFile)
+        {
+            /* If the source tree is as follows:
+             *     pkg/
+             *     +- package.d
+             *     +- common.d
+             * the 'pkg' will be incorporated to the internal package tree in two ways:
+             *     import pkg;
+             * and:
+             *     import pkg.common;
+             *
+             * If both are used in one compilation, 'pkg' as a module (== pkg/package.d)
+             * and a package name 'pkg' will conflict each other.
+             *
+             * To avoid the conflict:
+             * 1. If preceding package name insertion had occurred by Package::resolve,
+             *    later package.d loading will change Package::isPkgMod to PKGmodule and set Package::mod.
+             * 2. Otherwise, 'package.d' wrapped by 'Package' is inserted to the internal tree in here.
```

Comment:
I see `pkg/package.d` and `pkg/common.d`. But the comment references `Package`. Where is this name coming from and what does it mean? What does "wrapped by 'Package'" mean?